src/init.cpp

```diff
@@ -196,8 +196,9 @@ void Shutdown()
     StopRPC();
     StopHTTPServer();
 #ifdef ENABLE_WALLET
-    if (pwalletMain)
-        pwalletMain->Flush(false);
+    for (CWallet_ptr pwallet : vpwallets) {
```

Comment:
It's not in `wallet.cpp` now. Moving it there seems like a good idea, but independent of this PR...